File: JavaScript/3-abstraction.js
'use strict';

const power = Math.pow;
const square = x => power(x, 2);
const cube = x => power(x, 3);

console.log(power(10, 2)); // 100
console.log(square(10)); // 100

console.log(power(10, 3)); // 1000
console.log(cube(10)); // 1000

File: Perl_5/3-abstraction.pl
#!/usr/bin/perl

use strict;
use v5.10;

sub square {
	return shift() ** 2;
	# braces for perl to know that shift is called without args
}
sub cube {
	return shift() ** 3;
}
say 10 ** 2;
say square(10);

say 10 ** 3;
say cube 10; # demonstrating that no-braces feature again
